src/main/java/bio/terra/cli/service/utils/GoogleCloudStorage.java

```diff
@@ -13,37 +15,44 @@
   private static final Logger logger = LoggerFactory.getLogger(GoogleCloudStorage.class);
 
   // the Terra user whose credentials will be used to call authenticated requests
-  private final TerraUser terraUser;
+  private final GoogleCredentials googleCredentials;
 
   // the google project id of the workspace
   private final String googleProjectId;
 
   // the client object used for talking to Google Cloud Storage
-  private Storage storageClient;
+  private final Storage storageClient;
 
   /**
-   * Constructor for class that talks to Google Cloud Storage. The user must be authenticated.
-   * Methods in this class will use its credentials to call authenticated endpoints.
+   * Constructor for class that talks to Google Cloud Storage. Methods in this class will use the
+   * given credentials to call authenticated endpoints.
    *
-   * @param terraUser the Terra user whose credentials will be used to call authenticated endpoints
+   * <p>This constructor scopes Storage requests to the given project.
+   *
+   * @param googleCredentials the credentials that will be used to call authenticated endpoints
+   * @param googleProjectId scope Storage requests to this project
    */
-  public GoogleCloudStorage(TerraUser terraUser, String googleProjectId) {
-    this.terraUser = terraUser;
+  public GoogleCloudStorage(GoogleCredentials googleCredentials, String googleProjectId) {
+    this.googleCredentials = googleCredentials;
     this.googleProjectId = googleProjectId;
-    this.storageClient = null;
-    buildClientForTerraUser();
+    this.storageClient = buildClientForTerraUser();
   }
 
-  /** Build the GCS client object for the given Terra user. */
-  private void buildClientForTerraUser() {
+  /**
+   * Build the GCS client object for the given credentials and project.
+   *
+   * @return the GCS client object
+   */
+  private Storage buildClientForTerraUser() {
```

Comment:
`ForTerraUser` isn't as accurate now since it's just looking at the passed credentials. `buildClient` is probably fine.
Since this is being called only as a helper in the constructor, consider making this `static` and passing the required arguments directly. I don't think we need to keep the `googleCredentials` otherwise on this class.

Reply:
Good catch. Removed the `ForTerraUser` part of the method name and removed the `googleCredentials` as a property.